src/fsharp/CompileOps.fs

```diff
@@ -3412,13 +3412,14 @@ let ComputeAnonModuleName check defaultNamespace filename (m: range) =
 
 let PostParseModuleImpl (_i, defaultNamespace, isLastCompiland, filename, impl) = 
     match impl with 
-    | ParsedImplFileFragment.NamedModule(SynModuleOrNamespace(lid, isRec, isModule, decls, xmlDoc, attribs, access, m)) -> 
+    | ParsedImplFileFragment.NamedModule(SynModuleOrNamespace(lid, isRec, kind, decls, xmlDoc, attribs, access, m)) -> 
         let lid = 
             match lid with 
-            | [id] when isModule && id.idText = MangledGlobalName -> error(Error(FSComp.SR.buildInvalidModuleOrNamespaceName(), id.idRange))
+            | [id] when kind = NamedModule && id.idText = MangledGlobalName ->
```

Comment:
Should we be checking `kind.IsModule` instead of a `NamedModule` here?

Reply:
We aren't going to get `global` name in implicit module name, are we? 🙂
I can replace it to `IsModule` if it makes it more clear.

Reply:
I think it makes it more clear with `IsModule` because when you are specific with `NamedModule`, in my mind it implies: "`AnonModule` might have the global name". Also, it's more defensive and follows the same behavior as before.